src/app/scripts/cac/pages/cac-pages-home.js

```diff
@@ -306,12 +313,12 @@ CAC.Pages.Home = (function ($, ModeOptions,  MapControl, TripOptions, SearchPara
     // Note that components are responsible for doing the right thing based on whether they're
     // active or not.
     function onUrlChanged() {
-        setActiveTab();
-        modeOptionsControl.setMode(UserPreferences.getPreference('mode'));
-        directionsControl.setFromUserPreferences();
-        exploreControl.setFromUserPreferences();
         directionsFormControl.setFromUserPreferences();
+        directionsControl.setFromUserPreferences();
+        modeOptionsControl.setMode(UserPreferences.getPreference('mode'));
         showHideNeedWheelsBanner();
+        setActiveTab();
+        exploreControl.setFromUserPreferences();
```

Comment:
You're right, first doesn't make sense.  Now I'm thinking last does.  Explore and Directions will update based on user preferences but will only do their main thing (get directions or draw isochrone) if they're active.  Then when setActiveTab runs if the tab hasn't changed nothing will happen but if the tab has changed the one being activated will then do its thing.

The problem with that setup is that if the URL change implies a tab switch from Explore to Directions or vice versa then both actions will fire, and the one we're moving from will be firing uselessly.  Probably this is evidence that the triggering conditions are muddled and not quite right.  I.e. the logic of how/what to trigger on options/origin/destination changes vs URL changes.